openedx/features/lti_course_tab/tab.py

```diff
@@ -0,0 +1,253 @@
+        # lti_config = get_
+        pass
+
+    def get_lti_config(self, course: CourseDescriptor) -> LtiConfiguration:
+        return LtiConfiguration.objects.get(pk=self.lti_config_id)
+
+    def __init__(self, tab_dict=None, name=None, lti_config_id=None):
+        def link_func(course, reverse_func):
+            """ Returns a function that returns the static tab's URL. """
+            return reverse_func(self.type, args=[str(course.id), self.lti_config_id])
+
+        self.lti_config_id = tab_dict.get('lti_config_id') if tab_dict else lti_config_id
+
+        if tab_dict is None:
+            tab_dict = dict()
+
+        if name is not None:
+            tab_dict['name'] = name
+
+        tab_dict['link_func'] = link_func
+        tab_dict['tab_id'] = 'lti_tab_{0}'.format(self.lti_config_id)
+
+        super(LtiCourseTab, self).__init__(tab_dict)
+
+    @classmethod
+    def validate(cls, tab_dict, raise_error=True):
+        """
+        Ensures that the specified tab_dict is valid.
+        """
+        return (
+            super(LtiCourseTab, cls).validate(tab_dict, raise_error)
+            and key_checker(['name', 'lti_config_id'])(tab_dict, raise_error)
+        )
+
+    def __getitem__(self, key):
+        if key == 'lti_config_id':
+            return self.lti_config_id
+        else:
+            return super(LtiCourseTab, self).__getitem__(key)
+
+    def __setitem__(self, key, value):
+        if key == 'lti_config_id':
+            self.lti_config_id = value
+        else:
+            super(LtiCourseTab, self).__setitem__(key, value)
+
+    def to_json(self):
+        """ Return a dictionary representation of this tab. """
+        to_json_val = super(LtiCourseTab, self).to_json()
+        to_json_val.update({'lti_config_id': self.lti_config_id})
+        return to_json_val
+
+    def __eq__(self, other):
+        if not super(LtiCourseTab, self).__eq__(other):
+            return False
+        return self.lti_config_id == other.get('lti_config_id')
+
+    def __hash__(self):
+        """ Return a hash representation of Tab Object. """
+        return hash(repr(self))
+
+
+class LtiDiscussionCourseTab(LtiCourseLaunchMixin, EnrolledTab):
+    type = 'lti_discussion'
+    allow_multiple = False
+    is_dynamic = True
+    title = ugettext_lazy("LTI Discussions")
+
+    def __init__(self, tab_dict=None, name=None):
+        if tab_dict is None:
+            tab_dict = dict()
+
+        if name is not None:
+            tab_dict['name'] = name
+
+        super(LtiDiscussionCourseTab, self).__init__(tab_dict)
+
+    def get_lti_config(self, course: CourseDescriptor) -> LtiConfiguration:
+        config = DiscussionsConfiguration.objects.get(context_key=course.id)
+        return config.lti_configuration
+
+    @classmethod
+    def is_enabled(cls, course, user=None):
+        is_enabled = super().is_enabled(course, user)
+        if is_enabled:
+            try:
```

Comment:
With https://github.com/edx/edx-platform/pull/25913, this `try/except` block can go away:

```py
if is_enabled:
    config = DiscussionsConfiguration.get(course.id)
    return config.enabled and config.lti_configuration is not None
```

BTW: I _considered_ adding a `DiscussionsConfiguration.is_enabled_with_lti` helper in #25913 , but opted not to for now, thinking it was a little too specific. Happy to leave the LTI check here though.